src/transformers/utils/import_utils.py

```diff
@@ -484,6 +494,12 @@ def is_g2p_en_available():
 @lru_cache()
 def is_torch_tpu_available(check_device=True):
     "Checks if `torch_xla` is installed and potentially if a TPU is in the environment"
+    warnings.warn(
+        "`is_torch_tpu_available` is deprecated and will be removed in 4.39.0. "
```

Comment:
This will be the next release - so would need to be removed now! As it's a public object, it should go through at least two cycles

```suggestion
        "`is_torch_tpu_available` is deprecated and will be removed in 4.41.0. "
```

Reply:
Done